[ForTest]Expand procedure architecture for distributed execution, and support iceberg procedure `rewrite_data_files` (#22659)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

    CALL iceberg.system.expire_snapshots(schema => 'schema_name', table_name => 'table_name', snapshot_ids => ARRAY[10001, 10002]);

Rewrite Data Files
^^^^^^^^^^^^^^^^^^

> In Iceberg, we tracks all data files under different partition specs in a table. More data files leads to

Reviewer: ```suggestion
Iceberg tracks all data files under different partition specs in a table. More data files requires
```

> more metadata stored in manifest files, and small data files causes an unnecessary amount of metadata and

Reviewer: ```suggestion
more metadata to be stored in manifest files, and small data files can cause unnecessary amount metadata and
```

> less efficient queries from file open costs. Besides, data files under different partition specs might

Reviewer: ```suggestion
less efficient queries from file open costs. Also, data files under different partition specs can
```

> prevent metadata level deletion or predicate thoroughly push down for Presto engine.

Reviewer: ```suggestion
prevent metadata level deletion or thorough predicate push down for Presto.
```

> Procedure `rewrite_data_files` is used for rewriting the data files of the specified table so that they are

Reviewer: ```suggestion
Use `rewrite_data_files` to rewrite the data files of a specified table so that they are
```

merged into fewer but larger files under the newest partition spec. If the table is partitioned, the data

> files compaction could act separately on the selected partitions. This can improve read performance by reducing

Reviewer: ```suggestion
files compaction can act separately on the selected partitions to improve read performance by reducing
```

metadata overhead and runtime file open cost.

The following arguments are available:

===================== ========== =============== =======================================================================
Argument Name         required   type            Description
===================== ========== =============== =======================================================================

> ``schema``            ✔️         string          Schema of the table to update

Reviewer: ```suggestion
``schema``            ✔️         string          Schema of the table to update.
```

Because one sentence ("predicate as a string used for filtering the files.") requires punctuation, everything in this column does.

> ``table_name``        ✔️         string          Name of the table to update

Reviewer: ```suggestion
``table_name``        ✔️         string          Name of the table to update.
```

> ``filter``                       string          predicate as a string used for filtering the files. Note that currently

Reviewer: ```suggestion
``filter``                       string          Predicate as a string used for filtering the files. Currently
```
When you address these suggestions, please put the line breaks in the correct places for this entry for the table format. (To be clear: your draft formatting is great! I'm asking that you check to make sure my suggestions don't introduce a format error for the table.)

Author: Thanks for the reminder, got it.

>                                                  we only support rewrite whole partitions, that is, the filter should be

Reviewer: ```suggestion
                                                 only rewrite of whole partitions is  
```

>                                                  on partition columns(defaults to `true`)

Reviewer: ```suggestion
                                                 supported. Filter on partition 
                                                 columns. The default value is 
                                                 `true`.
```
